src/export/StructureDefinitionExporter.ts

```diff
@@ -637,30 +694,66 @@ export class StructureDefinitionExporter implements Fishable {
 
   applyDeferredRules() {
     this.deferredRules.forEach((rules, sd) => {
-      for (const rule of rules) {
+      for (const { rule, originalErr } of rules) {
+        let fishItem: string;
         if (typeof rule.value === 'string') {
-          const fishedValue = this.fishForFHIR(rule.value);
-          // An inline instance will fish up an InstanceDefinition, which can be used directly.
-          // Other instances will fish up an Object, which needs to be turned into an InstanceDefinition.
+          fishItem = rule.value;
+        } else if (
+          typeof rule.value === 'number' ||
+          typeof rule.value === 'bigint' ||
+          typeof rule.value === 'boolean'
+        ) {
+          fishItem = rule.rawValue;
+        }
+        let fishedValue = this.fishForFHIR(fishItem);
+        if (fishedValue) {
+          // If we are trying to assign a contained resource:
           // An InstanceDefinition of a resource needs a resourceType, so check for that property.
           // If we can't find a resourceType or an sdType, we have a non-resource Instance, which is no good
-          if (fishedValue) {
-            if (fishedValue instanceof InstanceDefinition && fishedValue.resourceType) {
-              try {
-                sd.setInstancePropertyByPath(rule.caretPath, fishedValue, this);
-              } catch (e) {
+          // if we can find the exported instance, it will have some useful metadata.
+          const instanceExporter = new InstanceExporter(this.tank, this.pkg, this.fisher);
+          const exportedInstance = instanceExporter.fishForFHIR(fishItem);
+          if (exportedInstance != null) {
+            fishedValue = exportedInstance;
+          }
```

Comment:
If the `InstanceExporter` returns a result, then you re-assign `fishedValue` without having ever used its original value. If the preferred source is the `InstanceExporter`, then wouldn't it be better to _start_ with that on line 708? And then if it returns null, fall back to `this.fishForFHIR` and (assuming it returns something), use `InstanceDefinition.fromJSON(fishedValue)` (see line 737) to turn it into an `InstanceDefinition`.

I might be missing something, but I feel like if you start at line 708 with:
```ts
const instanceExporter = new InstanceExporter(this.tank, this.pkg, this.fisher);
let fishedValue = instanceExporter.fishForFHIR(fishItem);
if (fishedValue == null) {
  const result = this.fishForFHIR(fishItem);
  if (!(result instanceof InstanceDefinition) && result instanceof Object) {
    fishedValue = InstanceDefinition.fromJSON(fishedValue);
  }
}
```
I feel like that might actually allow you to get rid of a lot of repeated code here. (But again, I may be overlooking something).